src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -1579,16 +1577,34 @@ def domain_unbind(cmd, client, resource_group, service, app, domain_name):
     return client.custom_domains.delete(resource_group, service, app, domain_name)
 
 
-def get_app_insights_key(cli_ctx, resource_group, name):
+def get_app_insights_credential_from_ai(appinsights):
+    if is_appinsights_with_bad_credential(appinsights):
+        return None
+    appInsightsCredential = ApplicationInsightsCredential()
+    appInsightsCredential.connection_string = appinsights.connection_string
+    appInsightsCredential.instrumentation_key = appinsights.instrumentation_key
+    return appInsightsCredential
+
+
+def get_app_insights_credential_by_name(cli_ctx, resource_group, name):
+    """Get connection_string first, if not exist, fall back to instrumentation_key.
+
+    :param name: name of Application Insights
+    """
     appinsights_client = get_mgmt_service_client(cli_ctx, ApplicationInsightsManagementClient)
     appinsights = appinsights_client.components.get(resource_group, name)
-    if appinsights is None or appinsights.instrumentation_key is None:
-        raise CLIError("App Insights {} under resource group {} was not found.".format(name, resource_group))
-    return appinsights.instrumentation_key
+    if is_appinsights_with_bad_credential(appinsights):
+        raise ResourceNotFoundError("App Insights {} under resource group {} was not found.".format(name, resource_group))
+    return get_app_insights_credential_from_ai(appinsights)
+
 
+def update_tracing_config_with_instrumentation_key_as_default(cmd, resource_group, service_name, location, app_insights_key,
```

Comment:
not sure if we can refine this method, which is similar to `update_java_agent_config_with_connection_string_as_default`.

Reply:
This method is only reference by `az spring-cloud update`, which will use `2020-07-01` api-version against `Resource Provider`, so I intend to make it clear. And indeed, it's a existed tech debt, since this method is very much similar to `update_java_agent_config_with_connection_string_as_default`. 

The benefit is, we can deco this method if we deco `az spring-cloud update --app-insights xxxx` command, and move to `az spring-cloud app-insights update --app-insights xxxxx`

Reply:
I see, then could you please help to add some comments here ? for that we will retired this method when GA, something similar stuff like that.

Reply:
Sure